packages/react/src/primitives/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,107 @@
+import classNames from 'classnames';
+import React from 'react';
+
+import { useCheckbox } from './useCheckbox';
+import { Flex } from '../Flex';
+import { IconCheck } from '../Icon';
+import { Input } from '../Input';
+import { Text } from '../Text';
+import { VisuallyHidden } from '../VisuallyHidden';
+import { CheckboxProps } from '../types/checkbox';
+import { ComponentClassNames } from '../shared/constants';
+import { useTestId } from '../utils/testUtils';
+
+export const Checkbox: React.FC<CheckboxProps> = ({
+  alignContent,
+  alignItems,
+  backgroundColor,
+  checked,
+  children,
+  className,
+  color,
+  defaultChecked,
+  direction,
+  hasError,
+  id,
+  isDisabled,
+  isReadOnly,
+  isRequired,
+  justifyContent,
+  gap,
+  name,
+  onChange: onChangeProp,
+  size,
+  testId,
+  value,
+  wrap,
+  ...rest
+}) => {
+  // controlled way should always override uncontrolled way
+  const initialChecked = checked !== undefined ? checked : defaultChecked;
+  const { dataChecked, dataFocus, onBlur, onChange, onFocus } = useCheckbox(
+    initialChecked,
+    onChangeProp
+  );
+
+  return (
+    <Flex
+      alignContent={alignContent}
+      alignItems={alignItems}
+      as="label"
+      className={classNames(ComponentClassNames.Checkbox, className)}
+      data-disabled={isDisabled}
+      direction={direction}
+      gap={gap}
+      justifyContent={justifyContent}
+      wrap={wrap}
+      testId={testId}
+    >
+      <VisuallyHidden>
+        <Input
+          checked={checked}
+          className={ComponentClassNames.CheckboxInput}
+          defaultChecked={defaultChecked}
+          id={id}
+          isDisabled={isDisabled}
+          isReadOnly={isReadOnly}
+          isRequired={isRequired}
+          name={name}
+          onBlur={onBlur}
+          onChange={onChange}
+          onFocus={onFocus}
+          type="checkbox"
+          value={value}
+          {...rest}
```

Comment:
I just realized that the style props will be useless here given this is VisuallyHidden. Is there a different element where they would make sense, perhaps on the root label? It's challenging to decide where the style props make sense on these custom styled primitives.

Reply:
Yea, it's hard to make a decision. We hide the really `input` here. Will move it up to container maybe a good choice